Hw security jwt

Petro-Smoliar opened the pull request.

Petro-Smoliar committed:
Added SecurityConfig class
Implemented UserDetails interface
Added the implementation for UserDetailsService interface
Added Role entity
Added Liquibase changesets

Petro-Smoliar committed:
Added SecurityConfig class
Implemented UserDetails interface
Added the implementation for UserDetailsService interface
Added Role entity
Added Liquibase changesets

Petro-Smoliar committed:
Added the following method to the AuthenticationController:
public UserLoginResponseDto login(UserLoginRequestDto request)
Added AuthenticationService, JwtAuthenticationFilter, JwtUtil classes

olekskov commented on src/main/java/com/example/book/store/model/User.java:
```diff
@@ -30,4 +38,46 @@ public class User {
     private String lastName;
     @Column(name = "shipping_addresses")
     private String shippingAddress;
+    @ManyToMany(fetch = FetchType.EAGER)
```
Using FetchType.EAGER is not the best practice.
Please use FetchType.LAZY and add fetch of roles on a query level

olekskov commented on src/main/java/com/example/book/store/repository/user/UserRepository.java:
```diff
@@ -1,7 +1,9 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 
 public interface UserRepository extends JpaRepository<User, Long> {
+    Optional<User> findByEmail(String email);
```
like this
```suggestion
    @EntityGraph(....
    Optional<User> findByEmail(String email);
```

Petro-Smoliar committed:
Added the following method to the AuthenticationController:
public UserLoginResponseDto login(UserLoginRequestDto request)
Added AuthenticationService, JwtAuthenticationFilter, JwtUtil classes
